libcomposefs/lcfs-mount.c

```diff
@@ -666,6 +668,35 @@ static errint_t lcfs_mount(struct lcfs_mount_state_s *state)
 	return -EINVAL;
 }
 
+static int lcfs_mount_copy_and_seal(int fd) {
+	int memfd = memfd_create(CFS_MOUNT_SOURCE, MFD_CLOEXEC | MFD_ALLOW_SEALING);
+	if (memfd < 0)
+		return -errno;
+
+	// in the general case, copy_file_range() to a memfd doesn't work
+	ssize_t n;
+	do {
+		char buffer[1 << 20];  // 1MB
+		n = read(fd, buffer, sizeof buffer);
+		if (n < 0)
+			break;
+		n = write(memfd, buffer, n);
```

Comment:
Need to handle short writes, EINTR and write errors in general. I mean, i'm sure memfd writes don't get i/o errors, but you may be low on memory or whatever.